apps/schema/src/components/TableView.vue

```diff
@@ -0,0 +1,311 @@
+            </thead>
+            <Draggable v-model="table.columns" tag="tbody" @end="applyPosition">
+              <ColumnView
+                v-for="(column, columnIndex) in table.columns"
+                :key="
+                  JSON.stringify(column) +
+                  '_' +
+                  columnIndex +
+                  '_' +
+                  table.columns.length
+                "
+                :style="
+                  isSubclassDropped(column)
+                    ? 'text-decoration: line-through'
+                    : ''
+                "
+                v-model="table.columns[columnIndex]"
+                :schema="schema"
+                :schemaNames="schemaNames"
+                @input="$emit('input', table)"
+                @createColumn="createColumn"
+                @add="addColumn(columnIndex, $event)"
+                @delete="deleteColumn(columnIndex)"
+                :isManager="isManager"
+              />
+            </Draggable>
+          </table>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<style>
+.hoverIcon {
+  visibility: hidden;
+}
+
+.hoverContainer:hover .hoverIcon {
+  visibility: visible;
+}
+</style>
+
+<script>
+import { IconDanger } from "molgenis-components";
+import columnTypes from "../columnTypes.js";
+import ColumnView from "./ColumnView.vue";
+import Draggable from "vuedraggable";
+import TableEditModal from "./TableEditModal.vue";
+import ColumnEditModal from "./ColumnEditModal.vue";
+
+export default {
+  components: {
+    TableEditModal,
+    ColumnView,
+    Draggable,
+    IconDanger,
+    ColumnEditModal,
+  },
+  props: {
+    value: {
+      type: Object,
+      required: true,
+    },
+    schema: {
+      type: Object,
+      required: true,
+    },
+    schemaNames: {
+      type: Array,
+      required: true,
+    },
+    isManager: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      table: {},
+      columnTypes,
+    };
+  },
+  methods: {
+    scrollToTop() {
+      this.$s;
+    },
+    addColumn(index, column) {
+      if (this.table.columns == undefined) {
+        this.table.columns = [];
+      }
+      this.table.columns.splice(index, 0, column);
+      this.$emit("input", this.table);
+    },
+    applyPosition() {
+      let position = 1;
+      this.table.columns.forEach((column) => (column.position = position++));
+      this.$emit("input", this.table);
+    },
+    validateName() {
+      if (!this.name) {
+        return "Table name is required";
+      }
+      if (this.schema.tables.filter((t) => t.name === this.name).length > 1) {
+        return "Table name must be unique within schema";
+      }
+    },
+    deleteTable(table) {
+      if (!table.oldName) {
+        this.$emit("delete");
+      } else if (!table.drop) {
+        //need to do deep set otherwise vue doesn't see it
+        this.$set(table, "drop", true);
+      } else {
+        this.$set(table, "drop", false);
```

Comment:
methods should not change there params, this effectively lead to manipulating variables in the global scope

Reply:
agreed, in other places I use and emit a clone. Will try to also apply here.